osu.Game/Rulesets/Edit/Tools/PlacementVisualiser.cs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Input;
+using osu.Framework.Input.EventArgs;
+using osu.Framework.Input.States;
+using osu.Framework.Timing;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets.Edit.Types;
+using osu.Game.Rulesets.Objects;
+using OpenTK;
+
+namespace osu.Game.Rulesets.Edit.Tools
+{
+    public abstract class PlacementVisualiser : CompositeDrawable, IRequireHighFrequencyMousePosition
+    {
+        public event Action<HitObject> PlacementFinished;
+
+        public readonly HitObject HitObject;
+
+        protected IBeatmap Beatmap { get; private set; }
+
+        private IAdjustableClock clock;
+
+        private double? lastAppliedTime;
+
+        protected PlacementVisualiser(HitObject hitObject)
+        {
+            HitObject = hitObject;
+
+            RelativeSizeAxes = Axes.Both;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(IBindableBeatmap workingBeatmap, IAdjustableClock clock)
+        {
+            this.clock = clock;
+
+            Beatmap = workingBeatmap.Value.Beatmap;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            // We'll only get a mouse move event in the next frame, so there's a 1-frame position discrepancy
+            // We can fix this by forcefully applying the current mouse position
+            ApplyPosition(GetContainingInputManager().CurrentState.Mouse.Position);
+        }
+
+        protected override void Update()
+        {
+            base.Update();
+
+            if (clock.CurrentTime != lastAppliedTime)
+            {
+                ApplyTime(clock.CurrentTime);
+                HitObject.ApplyDefaults(Beatmap.ControlPointInfo, Beatmap.BeatmapInfo.BaseDifficulty);
+
+                lastAppliedTime = clock.CurrentTime;
+            }
+        }
+
+        // Todo: Framework bug, this should not nullref
+        public override bool ReceiveMouseInputAt(Vector2 screenSpacePos) => Parent?.ReceiveMouseInputAt(screenSpacePos) ?? false;
```

Comment:
Not sure if this is intended functionality or if it's a bug, thoughts? maybe @ekrctb 

This drawable is getting removed from its parent on click, and it's receiving a double-click.